pkg/storage/chunk/storage/caching_index_client_test.go

```diff
@@ -45,22 +45,25 @@ func TestCachingStorageClientBasic(t *testing.T) {
 	require.NoError(t, err)
 	logger := log.NewNopLogger()
 	cache := cache.NewFifoCache("test", cache.FifoCacheConfig{MaxSizeItems: 10, Validity: 10 * time.Second}, nil, logger)
-	client := newCachingIndexClient(store, cache, 1*time.Second, limits, logger, false)
+	client := newCachingIndexClient(store, cache, 1*time.Second, limits, logger, false, 10, 100)
 	queries := []chunk.IndexQuery{{
 		TableName: "table",
 		HashValue: "baz",
 	}}
 	err = client.QueryPages(ctx, queries, func(_ chunk.IndexQuery, _ chunk.ReadBatch) bool {
 		return true
 	})
+
 	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
 	assert.EqualValues(t, 1, len(store.queries))
 
 	// If we do the query to the cache again, the underlying store shouldn't see it.
 	err = client.QueryPages(ctx, queries, func(_ chunk.IndexQuery, _ chunk.ReadBatch) bool {
 		return true
 	})
 	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
If I am not wrong, I think we do not need this sleep, the above suggested check for write back queue to be 0 should suffice.

Reply:
> If I am not wrong, I think we do not need this sleep, the above suggested check for write back queue to be 0 should suffice.

yes